File: 917-boats-to-save-people/boats-to-save-people.js
var numRescueBoats = function(people, limit) {
    weights = [...people].sort((a, b)=> a - b)

    let boats = 0

    i = 0
    j = people.length - 1

    while (i <= j) {
        boats += 1

        if (weights[i] <= limit - weights[j]) {
            i += 1
        }
        j -= 1
    }    

    return boats
};

File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        i = 0
        j = len(people) - 1
        boats = 0

        while (i <= j):
            diff = limit - people[j]

            if (diff >= people[i]):
                i += 1
            j -= 1
            boats += 1

        return boats